### PR TITLE
add update channel

### PR DESCRIPTION
**Pull request by channingduan**
add channel config update implementation testing code

see issue #30 
and update #55

### CODE REVIEW

File: pkg/identity/identity.go
Comment (SamYuan1990) [inline pkg/identity/identity.go:28]: could you please check if we have chance to refactor with `internal/pkg/identity/identity.go`

File: pkg/tools/tools.go
Comment (SamYuan1990) [inline pkg/tools/tools.go:173]: may I know what's magic value 0,0 here?

File: pkg/identity/identity.go
Comment (channingduan) [reply pkg/identity/identity.go:28]: > could you please check if we have chance to refactor with `internal/pkg/identity/identity.go`

ok, merge into Signer interface.

```go
type Signer interface {
	Sign(message []byte) ([]byte, error)
	Serialize() ([]byte, error)
}
```

File: pkg/tools/tools.go
Comment (channingduan) [reply pkg/tools/tools.go:173]: > may I know what's magic value 0,0 here?
change it. defind const value

File: pkg/identity/identity.go
Comment (bestbeforetoday) [inline pkg/identity/identity.go:27]: This should not be here. A client-supplied signing implementation should not need to know the internal Fabric mechanics of serialising an identity

File: pkg/identity/identity.go
Comment (bestbeforetoday) [inline pkg/identity/identity.go:84]: This should not need to be part of the public API

**Review by bestbeforetoday — CHANGES_REQUESTED** [review]

File: pkg/identity/identity.go
Comment (channingduan) [reply pkg/identity/identity.go:27]: ok. I will see how to modify it.

Comment (bestbeforetoday) [reply pkg/identity/identity.go:27]: The **Identity** interface already holds all the information we need to internally generate an **msp.SerializedIdentity** protobuf message:

```go
type Identity interface {
	MspID() string       // ID of the Membership Service Provider to which this identity belongs.
	Credentials() []byte // Implementation-specific credentials.
}
```

The code to create and serialize and **msp.SerializedIdentity** is very simple so we can either just do that explicitly in our code when we need the protobuf message, or we can have an _internal_ utility function to do that. There is similar code used in the internal Proposal creation already:

https://github.com/hyperledger/fabric-admin-sdk/blob/ea99b189b606915caabb0c339978757f55336b59/pkg/internal/proposal/transactioncontext.go#L31-L35

We know how to create the correct protobuf message. We should not be requiring the application code to do it for us.

File: pkg/identity/identity.go
Comment (channingduan) [reply pkg/identity/identity.go:83]: > This should not need to be part of the public API

the code here and Serialize function  has been removed.

**Comment by bestbeforetoday** [comment]
There is a build failure caused (I think) by a new static check release, not by this PR. I have merged a fix so you if you rebase on the current **main** branch, your PR should pass the build. Alternatively, you could review/merge #83 (which tidies up the build pipeline) and rebase on top of that.

**Comment by channingduan** [comment]
> There is a build failure caused (I think) by a new static check release, not by this PR. I have merged a fix so you if you rebase on the current **main** branch, your PR should pass the build. Alternatively, you could review/merge #83 (which tidies up the build pipeline) and rebase on top of that.
> 
> Edit: I've enabled required status checks for PRs that use the output from #83 so that change should go in before rebase.

OK~ I have rebase done,looks like it's ok now.

**Comment by SamYuan1990** [comment]
@bestbeforetoday , is this PR ready for merge?

**Comment by bestbeforetoday** [comment]
> @bestbeforetoday @channingduan , is this PR ready for merge?

The commit history of this PR looks to be a bit of a mess, and it is showing conflicts that would prevent a simple rebase on main. I will see if I can unpick some of the issues so we can get this and not hold up other PRs.

**Comment by bestbeforetoday** [comment]
I have rebased and squashed this change into a single commit on top of the current state of the **main** branch. I pushed that back to the PR branch so the **Files changed** for this PR now just shows the changes to be merged. There is a backup of exactly what was in the PR branch before I made these changes [here](https://github.com/bestbeforetoday/fabric-admin-sdk/tree/channingduan-feat/issues-30-backup).

**Review by bestbeforetoday — APPROVED** [review]
I have concerns that some of the content in `pkg/tools/tools.go`, which is part of the public API, expects internal structs as function parameters and returns internal types as function return values. Some of the functions (like `CreateSigner`) look more like test helper functions than things that should be part of the admin SDK public API, but that isn't something changed in this PR. Something to consider for future changes.

In general the change looks OK so I am happy for it to be merged. @SamYuan1990 it probably should have your review before merging.

**Review by SamYuan1990 — APPROVED** [review]
LGTM

**Pull request merged**
